packages/@aws-cdk/aws-appmesh/lib/mesh.ts

```diff
@@ -0,0 +1,368 @@
+import cdk = require('@aws-cdk/cdk');
+
+import { CfnMesh } from './appmesh.generated';
+import { ListenerProps, NAME_TAG } from './shared-interfaces';
+import { VirtualNode, VirtualNodeBackendProps } from './virtual-node';
+import { VirtualRouter, VirtualRouterBaseProps } from './virtual-router';
+import { VirtualService, VirtualServiceBaseProps } from './virtual-service';
+
+/**
+ * These properties are used when adding a VirtualNode through the Mesh type
+ */
+export interface AddVirtualNodeProps {
+  /**
+   * The name of the VirtualNode, only used as identifier
+   *
+   * @default optional if not provided, the id property will be used
+   */
+  readonly nodeName?: string;
+  /**
+   * The hostname for which to identify the VirtualNode, NOT the full FQDN
+   *
+   * @example serviceb NOT serviceb.domain.local
+   */
+  readonly hostname: string;
+  /**
+   * The service discovery namespace name
+   * @example domain.local
+   */
+  readonly namespaceName: string;
+  /**
+   * The backend services this node expects to send traffic to
+   *
+   * @default none
+   */
+  readonly backends?: VirtualNodeBackendProps[];
+  /**
+   * @default none
+   * if not specified must call addListeners(), addPortMappings() or addPortAndHealthCheckMappings()
+   */
+  readonly listeners?: ListenerProps;
+}
+
+/**
+ * A utility enum created for the egressFilter type property, the default of DROP_ALL,
+ * allows traffic only to other resources inside the mesh, or API calls to amazon resources.
+ *
+ * @default Drop_All
+ */
+export enum MeshFilterType {
+  /**
+   * Allows all outbound traffic
+   */
+  Allow_All = 'ALLOW_ALL',
```

Comment:
`ALLOW_ALL` or `AllowAll`, but not `Allow_All` (I find this a bit of an eyesore :D)

Reply:
Can't disagree there, just didn't know what convention to use. I like all caps for these types, will go with that if it works with you.